sale_order_deliverable_rate/models/sale_order.py

```diff
@@ -0,0 +1,35 @@
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    deliverable_rate = fields.Integer(
+        compute="_compute_deliverable_rate",
+        search="_search_deliverable_rate",
+        store=False,
+    )
+
+    def _compute_deliverable_rate(self):
+        for rec in self:
+            # filter order_lines who still need to be satisfies and count it
+            nb_order_line = len(
+                rec.order_line.filtered(lambda line: line.qty_to_ship > 0)
+            )
+            # if order_lines still need to be satisfies
+            # we make avg of their deliverable_rate
+            if nb_order_line > 0:
+                rec.deliverable_rate = (
+                    sum(rec.order_line.mapped("deliverable_rate")) / nb_order_line
+                )
+            else:
+                rec.deliverable_rate = 100
+
+    def _search_deliverable_rate(self, operator, value):
+        lines = self.env["sale.order.line"].search([("qty_to_ship", ">=", 0)])
```

Comment:
You can not search on no-stored field (cfr `sale.order.line` model).

And if you can use a more restrictive `domain`, it should be better for perf.